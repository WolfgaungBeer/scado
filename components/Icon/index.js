import React from 'react';
import { string, object } from 'prop-types';
import { withTheme } from 'styled-components';

const propTypes = {
    icon: string.isRequired,
    color: string,
    scale: string,
    theme: object.isRequired
};

const defaultProps = {
    color: undefined,
    scale: undefined
};

const Icon = ({ icon, color, scale, theme }) => {

    const themeColor = theme.color[color] || 'inherit';
    const themeScale = theme.scale[scale] || 'inherit';

    const style = {
        color: themeColor,
        fontSize: themeScale,
        transform: 'translateY(15%)'
    };

    return (<i className="material-icons" style={style}>{icon}</i>);
};

Icon.propTypes = propTypes;
Icon.defaultProps = defaultProps;

export default withTheme(Icon);
