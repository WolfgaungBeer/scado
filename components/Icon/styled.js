import styled, { keyframes, css } from 'styled-components';

const rotationKeyframes = keyframes`
    from {
        transform: rotate(0deg);
    }

    to {
        transform: rotate(360deg);
    }
`;

const StyledIcon = styled.i`
    ${(props) => {
        if (props.loading) {
            return css`
                display: inline-block;
                animation: ${rotationKeyframes} 0.85s linear infinite;
            `;
        }
        return '';
    }};
    color: ${props => props.theme.color[props.color] || 'inherit'};
    font-size: ${props => props.theme.scale[props.scale] || 'inherit'};
`;

export default StyledIcon;
