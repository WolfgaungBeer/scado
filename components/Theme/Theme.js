import React from 'react';
import { shape, object, node } from 'prop-types';
import { ThemeProvider } from 'styled-components';

const propTypes = {
    theme: shape({
        screen: object,
        color: object,
        scale: object,
    }).isRequired,
    children: node,
};

const defaultProps = {
    children: undefined,
};

const Theme = ({ theme, children }) => (<ThemeProvider theme={theme}>{children}</ThemeProvider>);

Theme.propTypes = propTypes;
Theme.defaultProps = defaultProps;

export default Theme;
