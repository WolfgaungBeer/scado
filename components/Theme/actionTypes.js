export const SET_THEME = '@SCADO/THEME/SET_THEME';
