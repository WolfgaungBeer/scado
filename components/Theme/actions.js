import types from './actionTypes';

const setTheme = payload => ({
    type: types.SET_THEME,
    payload,
});

export default {
    setTheme,
};
