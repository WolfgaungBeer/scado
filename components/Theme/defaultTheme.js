export default {

    color: {
        black: '#202021',
        white: '#F7F7F7',
        info: '#48C5FC',
        success: '#37B661',
        warning: '#DD9913',
        error: '#E25650'
    },

    scale: {
        xs: '0.25rem',
        s: '0.5rem',
        m: '1rem',
        l: '1.5rem',
        xl: '2rem',
        xxl: '2.5rem',
        xxxl: '3rem'
    }

}
