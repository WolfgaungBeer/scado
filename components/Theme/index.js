import { connect } from 'react-redux';
import { theme } from './selectors';
import Theme from './Theme';

const mapStateToProps = (state) => {
    return {
        theme: theme(state)
    };
};

export default connect(mapStateToProps)(Theme);
