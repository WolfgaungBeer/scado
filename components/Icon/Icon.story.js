import React from 'react';
import { storiesOf } from '@storybook/react';
import styled from 'styled-components';
import Theme from '../Theme/Theme';
import defaultTheme from '../Theme/defaultTheme';
import Text from '../Text';
import Icon from './';

import '../../node_modules/material-design-icons/iconfont/material-icons.css';

const IconWrapper = styled.div`
    color: #FF0FF0;
`;

storiesOf('Icon', module)
    .add('default', () => (
        <Theme theme={defaultTheme}>
            <Icon icon="group_work"/>
        </Theme>
    ))
    .add('color error and size l', () => (
        <Theme theme={defaultTheme}>
            <Icon icon="3d_rotation" color="error" scale="l" />
        </Theme>
    ))
    .add('color success and size xxl', () => (
        <Theme theme={defaultTheme}>
            <Icon icon="check_circle" color="success" scale="xxl" />
        </Theme>
    ))
    .add('wrapped in a Text.H1', () => (
        <Theme theme={defaultTheme}>
            <Text.H1><Icon icon="dashboard" />Icon Test</Text.H1>
        </Theme>
    ))
    .add('wrapped in a Text.H4', () => (
        <Theme theme={defaultTheme}>
            <Text.H4><Icon icon="dashboard" />Icon Test</Text.H4>
        </Theme>
    ))
    .add('wrapped in a Text.P', () => (
        <Theme theme={defaultTheme}>
            <Text.P><Icon icon="dashboard" />Icon Test</Text.P>
        </Theme>
    ))
    .add('wrapped in a Text.Label', () => (
        <Theme theme={defaultTheme}>
            <Text.Label><Icon icon="dashboard" />Icon Test</Text.Label>
        </Theme>
    ))
    .add('wrapped in a Text.Label with color and scale', () => (
        <Theme theme={defaultTheme}>
            <Text.Label color="success" scale="xxl"><Icon icon="dashboard" />Icon Test</Text.Label>
        </Theme>
    ))
    .add('custom icon', () => (
        <Theme theme={defaultTheme}>
            <IconWrapper>
                <Icon icon="dashboard" />
            </IconWrapper>
        </Theme>
    ));
