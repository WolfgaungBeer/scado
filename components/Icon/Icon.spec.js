import React from 'react';
import renderer from 'react-test-renderer';
import Theme from '../Theme/Theme';
import defaultTheme from '../Theme/defaultTheme';
import Icon from './';

it('renders the Icon Component correctly', () => {
    const tree1 = renderer.create(
        <Theme theme={defaultTheme}>
            <Icon icon="check_circle" />
        </Theme>,
    ).toJSON();
    expect(tree1).toMatchSnapshot();

    const tree2 = renderer.create(
        <Theme theme={defaultTheme}>
            <Icon icon="check_circle" color="success" scale="l" />
        </Theme>,
    ).toJSON();
    expect(tree2).toMatchSnapshot();

    const tree3 = renderer.create(
        <Theme theme={defaultTheme}>
            <Icon icon="check_circle" color="success" scale="l" loading />
        </Theme>,
    ).toJSON();
    expect(tree3).toMatchSnapshot();
});
