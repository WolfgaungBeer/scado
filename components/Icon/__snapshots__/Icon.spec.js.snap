// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Icon Component correctly 1`] = `
<i
  className="material-icons  sc-bdVaJa bhQYkA"
  color={undefined}
  scale={undefined}
>
  check_circle
</i>
`;

exports[`renders the Icon Component correctly 2`] = `
<i
  className="material-icons  sc-bdVaJa comCRk"
  color="success"
  scale="l"
>
  check_circle
</i>
`;

exports[`renders the Icon Component correctly 3`] = `
<i
  className="material-icons  sc-bdVaJa frNCAZ"
  color="success"
  scale="l"
>
  autorenew
</i>
`;
